runelite-client/src/main/java/net/runelite/client/plugins/pestcontrol/WidgetOverlay.java

```diff
@@ -199,6 +156,45 @@ private void renderWidgetOverlay(Graphics2D graphics, Portal portal, String text
 		graphics.drawString(text, x, y + 4);
 	}
 
+	private void renderProgressWidget(Graphics2D graphics)
+	{
+		String text;
+		int percentage;
+
+		Widget bar = client.getWidget(WidgetInfo.PEST_CONTROL_ACTIVITY_BAR).getChild(0);
+		Rectangle2D bounds = bar.getBounds().getBounds2D();
+
+		Widget prgs = client.getWidget(WidgetInfo.PEST_CONTROL_ACTIVITY_PROGRESS).getChild(0);
+
+		// At 0% the inner widget changes and your progress will not increase anymore
+		if ((int) (prgs.getBounds().getX()) - bounds.getX() != 2)
+		{
+			percentage = 0;
+			text = "FAILED";
+		}
+		else
+		{
+			percentage = (int) ((prgs.getBounds().getWidth() / bounds.getWidth()) * 100);
+			text = String.valueOf(percentage) + "%";
+		}
```

Comment:
Really love this change to replace the failed `6%` with `FAILED`, feels like a bugfix more than a feature addition like the rest of this PR. Would love to see this piece merged someday, even without the other changes this PR brings